staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -457,6 +458,54 @@ func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
 
+// fieldValidation checks that the field validation feature is enabled and if so
+// that the content type is supported.
+func fieldValidation(req *http.Request, s runtime.SerializerInfo) (string, error) {
+	var validationDirective string
+	if fv := req.URL.Query()["fieldValidation"]; len(fv) > 0 {
+		validationDirective = fv[0]
+	}
+	if validationDirective == metav1.FieldValidationWarn || validationDirective == metav1.FieldValidationStrict {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.StrictFieldValidation) {
+			return "", errors.NewBadRequest("the StrictFieldValidation feature is disabled")
+		}
+		if s.StrictSerializer == nil {
+			contentType := req.Header.Get("Content-Type")
+			return "", errors.NewBadRequest(fmt.Sprintf("fieldValidation parameter does not have a strict serializer for content type: %s", contentType))
+		}
+	}
+	return validationDirective, nil
+}
+
+// parseYAMLWarnings takes the strict decoding errors from the yaml decoder's output
+// and parses each individual warnings, or leaves the warning as is if
+// it does not look like a yaml strict decoding error.
+func parseYAMLWarnings(err error) []error {
```

Comment:
See my original [comment](https://github.com/kubernetes/kubernetes/pull/105916#discussion_r742195288) on this 

> Since applyPatch already disallows unknown fields, we are only checking for duplicate fields.
> 
> The way these duplicate fields get reported is when we `yaml.UnmarshalStrict` the patch to be applied. This uses a different mechanism than the serializer's json/yaml unmarshalling, and so it reports the error back in a different format.
> 
> The error that gets reported is a single large error with all the duplicate fields. Nested a few layers deep in it is a [yamlv2 TypeError](https://github.com/go-yaml/yaml/blob/v2.4.0/yaml.go#L270) which contains the individual errors that we would like to report back, but because the standard `goerrors.Unwrap` doesn't work on this error, the only way I've found to extract these errors is to do this ugly string parsing.
> 
> Let me know if there is a better way. Alternatively, we could just return it all as a single error (as done in the else statement) if we are okay with the duplicate field errors/warnings that come from applyPatch being different than those that come from the other endpoints.

Originally this yaml warning parsing was in the CRD handler. I moved it to the generic rest handlers package, because its used by all handlers anytime we get strict decoding errors from `yaml.YAMLToJSONStrict`. I still don't have a great solution to this. The choices I see are:
1. Regexp the yaml strict errors into something that looks more like the json strict errors. That's kind of what I've done here, but I could go even farther to make these errors look exactly like the json errors, but that requires even more regexp.
2. just remove the newline character and report all yaml strict errors as one large error. (we still need to remove the newline chars because attempting to write a warning header with a newline breaks and silently drops the whole warning message).
3. Figure out a better way of unwrapping the actual TypeError from the yaml library. This is probably the cleanest, but I haven't figured out how to do it yet.